vspace/src/main/java/edu/asu/diging/vspace/core/services/impl/SpaceManager.java

```diff
@@ -150,14 +158,35 @@ public ISpace getFullyLoadedSpace(String id) {
      *            if id is null throws exception, else delete corresponding
      *            space
      * @throws SpaceDoesNotExistException 
+     * @throws SpaceLinkDoesNotExistException 
      */
     @Override
-    public void deleteSpaceById(String id) throws SpaceDoesNotExistException {
+    public void deleteSpaceById(String id) throws SpaceDoesNotExistException, SpaceLinkDoesNotExistException {
         try {
             spaceRepo.deleteById(id);
-        } catch (IllegalArgumentException | EmptyResultDataAccessException exception) {
-            throw new SpaceDoesNotExistException(exception);
+            linkManager.deleteSpaceLinkBySource(id);
+        } catch (IllegalArgumentException | EmptyResultDataAccessException | SpaceLinkDoesNotExistException exception) {
+            if(exception instanceof SpaceLinkDoesNotExistException)
+                throw new SpaceLinkDoesNotExistException(exception);
+            else
+                throw new SpaceDoesNotExistException(exception);
```

Comment:
style guide

Reply:
I'm confused. Why do you wrap a SpaceLinkDoesNotExistException in a SpaceDoesNotExistException? The Space does exist, doesn't it?